src/main/java/core/basesyntax/service/ShopServiceImpl.java

```
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        transactions.forEach(transaction -> operationStrategy
+                        .get(transaction.getOperation())
+                        .apply(transaction));
```

process(...) must validate its input and handle errors with context. At minimum, check `transactions` for null and throw an informative RuntimeException (the checklist expects null-parameter validation). Also validate each element is non-null and that `transaction.getOperation()` is not null before calling operationStrategy.get(...). Consider wrapping `operationStrategy.get(...).apply(...)` in try-catch and rethrowing a RuntimeException that includes the transaction index/content to provide helpful debugging information instead of allowing raw NPEs or other exceptions to bubble without context. Quote from checklist: "Handle invalid inputs robustly by throwing RuntimeException with informative messages rather than printing or swallowing exceptions: ... Null parameters".